### PR TITLE
Likes history

### PR DESCRIPTION
**Pull request by Tanay-J**
This PR adds History and Like functionality
- User can now like/unlike a video, liked videos can be viewed in 'Likes' tab.
- Once a user clicks on a video to watch, the video gets added to History. History can be viewed and managed in 'History' tab.

### CODE REVIEW

File: src/components/history/history.module.css [hunk 1]
```diff
@@ -0,0 +1,56 @@
+.card_container {
+  display: grid;
+  grid-template-columns: 3rem 6rem auto auto;
+  grid-template-rows: auto;
+  grid-template-areas:
```
Comment (Varun-28) [inline src/components/history/history.module.css:5]: good use of grid template areas

File: src/utils/service-requests/history-services.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    dataDispatch({ type: "SET_HISTORY", payload: history });
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+const addToHistory = async (video, dataDispatch) => {
+  try {
+    const token = localStorage.getItem("token");
+    const {
```
Comment (Varun-28) [inline src/utils/service-requests/history-services.js:21]: - great and easy to understand the logic 👍

File: src/utils/service-requests/likes-services.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+const removeFromLikes = async (videoId, dataDispatch) => {
+  try {
+    const token = localStorage.getItem("token");
+    const {
+      data: { likes },
+    } = await axios.delete(`/api/user/likes/${videoId}`, {
+      headers: { authorization: token },
+    });
+    dataDispatch({ type: "SET_LIKES", payload: likes });
```
Comment (Varun-28) [inline src/utils/service-requests/likes-services.js:43]: - proper file segmentation and code formatting 👍

File: src/components/history/history.module.css [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+}
+ul {
+  list-style-type: none;
+  padding: 4px;
+}
+.options {
+  position: absolute;
+  right: 5%;
+  bottom: 0px;
+  background-color: rgba(66, 65, 65, 0.9);
```
Comment (vijaytembugade) [inline src/components/history/history.module.css:53]: CSS Variable would be prefered here!

File: src/contexts/data-context.js [hunk 5]
```diff
@@ -4,7 +4,12 @@ const DataContext = createContext();
 const useData = () => useContext(DataContext);
 
 const DataProvider = ({ children }) => {
-  const initialState = { watchlater: [], playlists: [], history: [] };
+  const initialState = {
+    watchlater: [],
+    playlists: [],
+    history: [],
+    likes: [],
```
Comment (vijaytembugade) [inline src/contexts/data-context.js:11]: Suggestion : I am not sure it is good to write everything in same context. You can create different contenxt for each

**Review by vijaytembugade — APPROVED** [review]
UI is cool. 🔥
suggestion : Use barrel export

File: src/components/history/history.module.css (hunk 4 above)
Comment (Tanay-J) [reply src/components/history/history.module.css:53]: Hi Vijay, this is one off usage, so I did not create a variable.

File: src/contexts/data-context.js (hunk 5 above)
Comment (Tanay-J) [reply src/contexts/data-context.js:11]: Even I was of two minds here. Will read more on how this affects app performance and make necessary changes accordingly.

**Pull request merged**